index.spec.js

```diff
@@ -0,0 +1,77 @@
+var { beforeAll, describe, expect, test } = require('@jest/globals');
+var { GenericContainer, Wait } = require('testcontainers');
+var Client = require('./index').Client;
+
+test('cannot use basic and custom auth', function(){
+  expect(function() {
+    new Client({
+      host: 'localhost',
+      port: 8080,
+      basic_auth: {
+        username: 'test',
+        password: 'test',
+      },
+      custom_auth: 'Token abc',
+    });
+  }).toThrow(new Error('Please do not specify basic_auth and custom_auth at the same time.'));
+});
+
+describe.each([
+  ['presto', 'ahanaio/prestodb-sandbox:0.281'],
+  ['trino', 'trinodb/trino:418'],
+])('%s', function(engine, image){
+  var container;
+  var client;
+  beforeAll(function(done){
+    new GenericContainer(image)
```

Comment:
I waffled between using docker-compose vs testcontainers and settled on testcontainers as it gives more flexibility in terms of setting up tests that require some custom server configuration (e.g. like basic auth), though that is still all possible with docker-compose.

The runtime cost of using testcontainers is that running the test suite takes ~30s, with most of that time taken purely on waiting for the containers to start.

If opposed, easy enough to switch to using docker-compose, let me know what you'd prefer @tagomoris.

Reply:
I'm totally fine with the library selection 👍